lib/grunt/file.js

```diff
@@ -199,6 +199,11 @@ file.mkdir = function(dirpath, mode) {
       try {
         fs.mkdirSync(subpath, mode);
       } catch(e) {
+        if(e.code === 'EEXIST' && file.isDir(subpath)) {
+          // The path was created between exists() and mkdirSync().
+          // If it's a directory, we're done.
```

Comment:
Also, if `mode` is passed in and the directory doesn't have the same `mode` should it adjust the permissions or no?